### PR TITLE
allow header injection for html node

### PR DESCRIPTION
**Pull request by fakob**
* added a header socket which allows to add code in the header of the html node
  * as default example we are loading tailwind and style the text with it

Screenshot 2023-12-06 at 22 35 46

### CODE REVIEW

File: src/nodes/draw/html.tsx [hunk 1]
```diff
@@ -101,47 +111,59 @@ export class HtmlRenderer extends HybridNode2 {
 
   // small presentational component
   protected getParentComponent(props: any): React.ReactElement {
-    const node = props.node;
-    const iframeRef = useRef();
-    const [htmlData, setHtmlData] = useState(props[inputSocketName]);
-    const [reload, setReload] = useState(props[reloadSocketName]);
-
-    useEffect(() => {
-      if (iframeRef.current) {
```
Comment (magnificus) [inline src/nodes/draw/html.tsx:110]: when would this be invalid? can check be removed?

File: src/nodes/draw/html.tsx [hunk 2]
```diff
@@ -190,7 +212,7 @@ export class EmbedWebsite extends HtmlRenderer {
   protected getDefaultIO(): PPSocket[] {
     const codeSocket = super
       .getDefaultIO()
-      .find((socket) => socket.name === inputSocketName);
+      .find((socket) => socket.name === inputSocketNameHtml);
```
Comment (magnificus) [inline src/nodes/draw/html.tsx:215]: You can do this much more elegantly with inheritance, change defaultCode in the parent to

getDefaultHTMLCode() (or some other name), then in parent getDefaultIO

      new PPSocket(
        SOCKET_TYPE.IN,
        inputSocketNameHtml,
        new CodeType(),
        getDefaultHTMLCode(),
        false,

then in child just override

getDefaultHTMLCode() with
return `<iframe src="https://en.wikipedia.org/wiki/Special:Random" style="width: 100%; height: 100%;"></iframe>`

**Review by magnificus — CHANGES_REQUESTED** [review]

File: src/nodes/draw/html.tsx (hunk 2 above)
Comment (fakob) [reply src/nodes/draw/html.tsx:215]: Good idea!

File: src/nodes/draw/html.tsx (hunk 1 above)
Comment (fakob) [reply src/nodes/draw/html.tsx:110]: Good question. I think with the constructor update this is no longer an issue. Will remove.

**Review by magnificus — APPROVED** [review]

**Pull request merged**